### PR TITLE
Fix orogen create for debian packaging

### PR DESCRIPTION
**Pull request by 2maz**
Move project creation templates into the template folder (under lib/) and provide a robust way to identify the library directory.
This allows for a working installation using gem2deb

### CODE REVIEW

File: lib/orogen/gen/templates.rb [hunk 1]
```diff
@@ -9,10 +9,22 @@ class << self
 	    attr_reader :templates
 	end
 
-        # Returns the directory where Orogen's lib part sits (i.e. where
-        # autobuild.rb and autobuild/ are)
+        # Returns the directory where oroGen's lib part sits, i.e.
+        # base_dir is defined by the directory where orogen.rb resides
         def self.base_dir
```
Comment (doudou) [inline lib/orogen/gen/templates.rb]: Oooookaaaay.

Can you detail how gen/../.. is not lib under gem2deb ????

Comment (2maz) [inline lib/orogen/gen/templates.rb]: Generally, with the packaging everything under lib will go into lib/ruby/vendor_ruby/ (so there is no 'lib' directory, but the main reference is orogen.rb). 
When moving the templates under lib/ then this is not necessary for this particular case.
Also, if I can assume that there are no further references relative to and going outside of lib/ (e.g. by plugins) then this fix is obsolete.
The documentation should still change as suggested.

Comment (doudou) [inline lib/orogen/gen/templates.rb]: I'm not sure I'm following. `base_dir` was returning the place of the contents of `lib/` isn't it ? Isn't that the same than the contents of `vendor_ruby`, i.e. `File.join(base_dir, 'orogen')` would be true in both setups. What am I missing ?

The problem as far as I understand it was that the templates were trying to go *out* of lib/ which is not possible. The fix you've applied (moving them into lib/) is definitely the right one.

> Also, if I can assume that there are no further references relative to and going outside of lib/ (e.g. by plugins) then this fix is obsolete.

That's definitely what we should enforce that.

Comment (2maz) [inline lib/orogen/gen/templates.rb]: not missing anything but my confused state of mind ;), so sticking with moving the templates (and leaving out the change of base_dir, except for the documentation change)

**Pull request merged**